prometheus/promhttp/http.go

```diff
@@ -406,3 +418,49 @@ func httpError(rsp http.ResponseWriter, err error) {
 		http.StatusInternalServerError,
 	)
 }
+
+func GetWriter(r *http.Request, rsp http.ResponseWriter, disableCompression bool, offeredCompressions []Compression) (io.Writer, error) {
```

Comment:
(non blocking suggestion): To improve reliability of this function I would suggest returning headers callers will set (on success) instead of modifying `http.ResponseWriter` (reduce side effects). To ensure this I would also change rsp to io.Writer, so:

```suggestion
func NegotiateEncodingWriter(r *http.Request, rw io.Writer, disableCompression bool, offeredCompressions []Compression) (_ io.Writer, encodingHeaderValue string, _ error) {
```